$OpenBSD$

Adjust the HTTP/2 client callback signature for the `http2-5.4.0` API used
on OpenBSD and import the helper that the newer API expects.

--- src/Simplex/Messaging/Transport/HTTP2/Client.hs.orig
+++ src/Simplex/Messaging/Transport/HTTP2/Client.hs
@@ -27,6 +27,7 @@
 import Control.Exception (IOException, try)
 import qualified Control.Exception as E
 import Control.Monad
+import qualified Data.ByteString.Char8 as B
 import Data.Functor (($>))
 import Data.Time (UTCTime, getCurrentTime)
 import qualified Data.X509 as X
@@ -146,7 +147,7 @@
         Nothing -> cancel action $> Left (HCNetworkError NETimeoutError)

     client :: HClient -> TMVar (Either HTTP2ClientError HTTP2Client) -> TLS p -> H.Client HTTP2Response
-    client c cVar tls sendReq = do
+    client c cVar tls sendReq _aux = do
       sessionTs <- getCurrentTime
       let c' =
             HTTP2Client
@@ -164,10 +165,10 @@
       atomically $ do
         writeTVar (connected c) True
         putTMVar cVar (Right c')
-      process c' sendReq `E.finally` disconnected
+      process c' sendReq _aux `E.finally` disconnected

     process :: HTTP2Client -> H.Client HTTP2Response
-    process HTTP2Client {client_ = HClient {reqQ}} sendReq = forever $ do
+    process HTTP2Client {client_ = HClient {config, reqQ}} sendReq _aux = forever $ do
       (req, respVar) <- atomically $ readTBQueue reqQ
       sendReq req $ \r -> do
         respBody <- getHTTP2Body r (bodyHeadSize config)
@@ -213,4 +214,4 @@
 runHTTP2ClientWith bufferSize host setup client = setup $ \tls -> withHTTP2 bufferSize (run tls) (pure ()) tls
   where
     run :: TLS p -> H.Config -> IO a
-    run tls cfg = H.run (ClientConfig "https" (strEncode host) 20) cfg $ client tls
+    run tls cfg = H.run (H.defaultClientConfig {H.scheme = "https", H.authority = B.unpack $ strEncode host, H.cacheLimit = 20}) cfg $ client tls
